plugins/inputs/gnmi/handler.go

```diff
@@ -241,12 +242,16 @@ func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, respon
 				key = parts[1]
 			}
 		} else {
-			if len(aliasPath) < len(key) && len(aliasPath) != 0 {
-				// This may not be an exact prefix, due to naming style
-				// conversion on the key.
-				key = key[len(aliasPath)+1:]
-			} else if len(aliasPath) >= len(key) {
-				// Otherwise use the last path element as the field key.
+			if aliasInfo.origin == field.path.origin {
+				if len(aliasPath) < len(key) && len(aliasPath) != 0 {
+					// This may not be an exact prefix, due to naming style
+					// conversion on the key.
+					key = key[len(aliasPath)+1:]
+				} else if len(aliasPath) >= len(key) {
+					// Otherwise use the last path element as the field key.
+					key = path.Base(key)
+				}
+			} else {
 				key = path.Base(key)
 			}
```

Comment:
Yeah this doesn't work straight away, as the alias path shows up with an extra leading slash:

```s
field.path:  type:/model/some/path
aliasInfo:   type://model
```

Reply:
We should strip those empty ones out in the first place when constructing the path I guess. It might also be necessary to do the comparison case-insensitive...

Reply:
I briefly look at this, it is the call to `normalize()` that does this. I need to go understand that function a bit more.

Reply:
ok figured it out, update pushed